library/core/src/str/mod.rs

```diff
@@ -212,7 +208,81 @@ impl str {
             None => index == self.len(),
 
             // This is bit magic equivalent to: b < 128 || b >= 192
-            Some(&b) => (b as i8) >= -0x40,
+            Some(&b) => b.is_utf8_char_boundary(),
+        }
+    }
+
+    /// Finds the closest `x` not exceeding `index` where `is_char_boundary(x)` is `true`.
```

Comment:
Just a suggestion...

```suggestion
    /// Finds the nearest char boundary at or below `index`.
    ///
    /// See the [is_char_boundary] method.
```